docs/conf.py

```diff
@@ -297,11 +298,78 @@
 # Auto-generation of documentation
 ##################################
 
+def _version_from_version_string(version):
+    """Get a tuple of version from a juju version string.
+
+    Note that release of juju use Major.Minor.Patch (eg, 3.6.0) but releases will use Major.Minor-betaBeta.
+    This returns a tuple of either (3,6,0) or (3, 6, 'beta5').
+    If neither can be found, this returns None
+    """
+    version_re = re.compile('(?P<major>\d+)\.(?P<minor>\d+)(\.(?P<patch>\d+)|-(?P<beta>beta\d+))')
+    m = version_re.match(version)
+    if m is None:
+        return None
+    patch = m['patch']
+    if patch is None:
+        patch = m['beta']
+    else:
+        patch = int(patch)
+    return (int(m['major']), int(m['minor']), patch)
+
+def _extract_version_from_version_go(version_file):
+    """Extract the version string from a juju version.go file"""
+    # Note that this assumes Major and Minor are integers, but patch might be an integer or something like beta5.
+    version_re = re.compile('const version = "(?P<version>[^"]*)"')
+    for line in version_file:
+        m = version_re.match(line)
+        if m is None:
+            continue
+        return _version_from_version_string(m['version'])
+    return None
+
+
+def get_tree_juju_version():
+    """Read the version of juju as reported by the juju branch that we're building."""
+    try:
+        # This is the location in Juju 3.6
+        with open('../version/version.go', 'rt') as version_file:
+            version = _extract_version_from_version_go(version_file)
+    except FileNotFoundError:
+        # This is the location in Juju 4.0
+        with open('../core/version/version.go', 'rt') as version_file:
+            version = _extract_version_from_version_go(version_file)
+    if version is None:
+        raise RuntimeError("could not determine the version of Juju for this directory")
+    return version
+
+
+def get_juju_version():
+    """Check to see what version of Juju we are running."""
+    # There is probably more that could be done here about all the possible
+    # version strings juju spits out, but this should cover stripping things
+    # like 'genericlinux' and the architecture out.
+    version_re = re.compile('(?P<version>.*)-genericlinux.*')
```

Comment:
Doesn't work on my machine :(

But does with this change:

```suggestion
    version_re = re.compile('(?P<version>.*)-(genericlinux|amd64).*')
```

Reply:
Maybe good to add arm there as well for all those folks building Juju on the RPis 🙃

Reply:
I'm guessing yours is actually `3.6.3-ubuntu-amd64` and the issue is `ubuntu` not `genericlinux`, but I have accommodated both by dropping all the way down to major.minor and reworking the parsing a bit.
Please do another quick QA to make sure it works correctly, but I see it refusing juju from snap, but allowing juju 3.6.2 or 3.6.3 to build the source.